tools/distributions-scraper/main.py

```diff
@@ -0,0 +1,106 @@
+import importlib
+import pkgutil
+import asyncio
+import json
+import pathlib
+import sys
+import logging
+from concurrent.futures import ThreadPoolExecutor
+from scraper.base import BaseScraper
+import scraper
+
+
+DEFAULT_OUTPUT_FILE = (
+    pathlib.Path(__file__).resolve().parent.parent.parent
+    / "data"
+    / "distributions"
+    / "distribution-info.json"
+)
+
+logger = logging.getLogger(__name__)
+
+
+def configure_logging(level: int = logging.INFO):
+    """Configure root logger for CLI usage."""
+    logging.basicConfig(
+        level=level,
+        format="%(asctime)s %(levelname)s %(name)s: %(message)s",
+        stream=sys.stdout,
+    )
+
+def load_scrapers():
+    """Dynamically load all scraper classes from scraper/ package."""
+    scrapers = []
+    for _, module_name, _ in pkgutil.iter_modules(scraper.__path__):
+        if module_name == "base":
+            continue
+
+        module = importlib.import_module(f"scraper.{module_name}")
+        for attr_name in dir(module):
+            obj = getattr(module, attr_name)
+            if isinstance(obj, type) and issubclass(obj, BaseScraper) and obj is not BaseScraper:
+                scrapers.append(obj())
+    return scrapers
+
+
+def write_output_file(output, path = DEFAULT_OUTPUT_FILE):
+    """Write JSON output to the given path, creating parent directories as needed."""
+    path.parent.mkdir(parents=True, exist_ok=True)
+    json_str = json.dumps(output, indent=2, sort_keys=True) + "\n"
+    path.write_text(json_str)
```

Comment:
Not sure how large the JSON data is, but we could dump to the file directly, which might save some memory and feels more idiomatic to me:
```suggestion
    with path.open('w') as f:
        json.dump(output, f, indent=2, sort_keys=True)
        f.write('\n')
```